### PR TITLE
Finishes migration to testing/testify

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Per #130, this finishes the port of tests to testing/testify.

The only thing pulling in ginkgo/gomega now is the istio code. This may be obviated after #128 

```
$  go mod why github.com/onsi/ginkgo
# github.com/onsi/ginkgo
github.com/tetratelabs/getenvoy/pkg/binary/envoy/controlplane
github.com/tetratelabs/getenvoy/pkg/binary/envoy/controlplane.test
istio.io/istio/pilot/pkg/bootstrap
istio.io/istio/pkg/util
k8s.io/apimachinery/pkg/util/strategicpatch
k8s.io/kube-openapi/pkg/util/proto
k8s.io/kube-openapi/pkg/util/proto.test
github.com/onsi/ginkgo

$ go mod why github.com/onsi/gomega 
# github.com/onsi/gomega
github.com/tetratelabs/getenvoy/pkg/binary/envoy/controlplane
istio.io/istio/pkg/config/mesh
istio.io/istio/pkg/config/mesh.test
github.com/onsi/gomega
```

### CODE REVIEW

File: pkg/util/os/signals_test.go [hunk 1]
```diff
@@ -18,136 +18,158 @@ import (
...
-
-				By("sending second relevant signal to the process")
-				Expect(syscall.Kill(syscall.Getpid(), relevantSignal)).To(Succeed())
-				Expect(stopCh).To(BeClosed())
-				Consistently(terminateCh).ShouldNot(BeClosed())
-			})
-		})
-	})
-})
+func TestShutdownSignals(t *testing.T) {
```
Comment (codefromthecrypt) [inline pkg/util/os/signals_test.go:27]: this file has the only interesting tests in it

File: pkg/util/os/signals_test.go [hunk 2]
```diff
@@ -18,136 +18,158 @@ import (
...
+	defer cancel()
+
+	stopCh := SetupSignalHandler(ctx)
+	cancel()
+
+	// send a relevant signal to the process
+	err := syscall.Kill(syscall.Getpid(), syscall.SIGINT)
+	require.NoError(t, err)
+
+	// First relevant signal closes the channel, but doesn't terminate
```
Comment (codefromthecrypt) [inline pkg/util/os/signals_test.go:57]:
```suggestion
	// The signal is ignored because the context closed prior to receiving it
```

**Comment by codefromthecrypt** [comment]
kicked test as e2e failed due to known issue with old macOS+rust script

**Comment by mathetake** [comment]
let me review tomorrow..

File: go.mod [hunk 3]
```diff
@@ -22,8 +22,8 @@ require (
 	github.com/mattn/go-shellwords v1.0.10
 	github.com/mholt/archiver v3.1.1+incompatible
 	github.com/mitchellh/go-homedir v1.1.0
-	github.com/onsi/ginkgo v1.12.1
-	github.com/onsi/gomega v1.10.0
+	github.com/onsi/ginkgo v1.12.1 // indirect
+	github.com/onsi/gomega v1.10.0 // indirect
```
Comment (mathetake) [inline go.mod:26]: 🎉

File: pkg/util/cmd/run.go [hunk 4]
```diff
@@ -55,9 +55,9 @@ func (fn ComposableRunE) Then(nextFn Run) ComposableRunE {
 // Using this function it is possible to compose more sophisticated behavior, e.g. one
 // where a handler function is called on multiple commands.
 func CallParentPersistentPreRunE() ComposableRunE {
-	return ComposableRunE(func(cmd *cobra.Command, args []string) error {
+	return func(cmd *cobra.Command, args []string) error {
 		return persistentPreRunE(cmd.Parent(), args)
-	})
+	}
```
Comment (mathetake) [inline pkg/util/cmd/run.go:60]: just curious, what is this change about?

File: pkg/extension/workspace/example/runtime/configdir/configdir_test.go [hunk 5]
```diff
@@ -19,173 +19,184 @@ import (
...
+func expectValidBootstrap(t *testing.T, bootstrap *envoybootstrap.Bootstrap) {
+	require.NotNil(t, bootstrap)
+	require.Equal(t, "/dev/null", bootstrap.GetAdmin().GetAccessLogPath())
+	require.Equal(t, "127.0.0.1", bootstrap.GetAdmin().GetAddress().GetSocketAddress().GetAddress())
+	require.Equal(t, uint32(9901), bootstrap.GetAdmin().GetAddress().GetSocketAddress().GetPortValue())
+}
+
+func expectInvalidBootstrap(t *testing.T, bootstrap *envoybootstrap.Bootstrap) {
+	require.Nil(t, bootstrap)
+}
```
Comment (mathetake) [inline pkg/extension/workspace/example/runtime/configdir/configdir_test.go:153]: Can we just inline this since it seems just aliasing `require.Nil`?

**Review by mathetake — APPROVED** [review]

File: pkg/util/cmd/run.go (hunk 4 above)
Comment (codefromthecrypt) [reply pkg/util/cmd/run.go:60]: IDE warning that it was redundant

File: pkg/extension/workspace/example/runtime/configdir/configdir_test.go (hunk 5 above)
Comment (codefromthecrypt) [reply pkg/extension/workspace/example/runtime/configdir/configdir_test.go:153]: sgtm!

Comment (codefromthecrypt) [reply pkg/extension/workspace/example/runtime/configdir/configdir_test.go:153]: actually it is messier due to `, msgAndArgs ...interface{}` which isn't entirely visible here. leaving as-is, but thanks for the suggestion as was worthwhile looking

**Pull request merged**